clap_generate/src/generators/shells/zsh.rs

```diff
@@ -475,28 +476,45 @@ fn write_opts_of(p: &App) -> String {
     ret.join("\n")
 }
 
-fn arg_conflicts(app: &App, arg: &Arg) -> String {
-    let conflicts = app.get_arg_conflicts_with(arg);
+fn arg_conflicts(app: &App, arg: &Arg, app_global: Option<&App>) -> String {
+    fn push_conflicts(conflicts: &[&Arg], res: &mut Vec<String>) {
+        for conflict in conflicts {
+            if let Some(s) = conflict.get_short() {
+                res.push(format!("-{}", s));
+            }
 
-    if conflicts.is_empty() {
-        return String::new();
+            if let Some(l) = conflict.get_long() {
+                res.push(format!("--{}", l));
+            }
+        }
     }
 
     let mut res = vec![];
-    for conflict in conflicts {
-        if let Some(s) = conflict.get_short() {
-            res.push(format!("-{}", s));
+    match (app_global, arg.get_global()) {
```

Comment:
Let's decrease the match scope to only get the `conflicts`. That way you can reduce duplication and don't even need the `push_conflicts` function.

Reply:
Decreasing the scope is problematic because `app_global` and `app` have different `lifetimes`. If I try to handle the `conflicts` outside the `match` I get  a `lifetime mismatch error`. Because this isn't performance critical and the duplication isn't that large, I would ask you, if this change is important enough, for me to take a longer look at this again or if the merge can proceed?